spec/unit/provider/service/smf_spec.rb

```diff
@@ -174,14 +176,35 @@
   end
 
   describe "when restarting" do
-    it "should call 'svcadm restart /system/myservice'" do
-      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "/system/myservice"], true)
-      @provider.expects(:wait).with('online')
-      @provider.restart
+    context 'with :kernelrelease == 5.10' do
+      it "should call 'svcadm restart /system/myservice'" do
+        Facter.clear
+        Facter.stubs(:fact).with(:kernelrelease).returns Facter.add(:kernelrelease) { setcode { '5.10' } }
+        @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "/system/myservice"], true)
+        @provider.expects(:wait).with('online')
+        @provider.restart
+      end
+    end
+    context 'with :kernelrelease == 5.11' do
+      it "should call 'svcadm restart -s /system/myservice'" do
+        @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "-s", "/system/myservice"], true)
+        @provider.expects(:wait).with('online')
+        @provider.restart
+      end
+    end
+    context 'with :kernelrelease > 5.11' do
+      it "should call 'svcadm restart -s /system/myservice'" do
+        Facter.clear
+        Facter.stubs(:fact).with(:kernelrelease).returns Facter.add(:kernelrelease) { setcode { '5.12' } }
+        @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "-s", "/system/myservice"], true)
+        @provider.expects(:wait).with('online')
+        @provider.restart
+      end
     end
 
     it "should error if timeout occurs while restarting the service" do
-      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "/system/myservice"], true)
+      Facter.loadfacts # With facts cleared timeout testing fails; load them again
```

Comment:
Ah, because you're resetting facts earlier. You probably still need to do something to resolve facts before setting the expectation on Timeout.